ui-v2/app/controllers/dc/acls/policies/edit.js

```diff
@@ -0,0 +1,38 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { get, set } from '@ember/object';
+export default Controller.extend({
+  builder: service('form'),
+  dom: service('dom'),
+  isScoped: false,
+  setProperties: function(model) {
+    this.form = get(this, 'builder')
+      .form('policy')
+      .setData(model.item);
+    // essentially this replaces the data with changesets
+    this._super({
+      ...model,
+      ...{
+        item: this.form.getData(),
+      },
+    });
+    set(this, 'isScoped', get(model.item, 'Datacenters.length') > 0);
+  },
+  actions: {
+    change: function(e, value, item) {
+      try {
+        get(this, 'form').handleEvent(get(this, 'dom').normalizeEvent(e, value));
+      } catch (e) {
+        const target = e.target || { name: null };
+        switch (target.name) {
+          case 'policy[isScoped]':
+            set(this, 'isScoped', !get(this, 'isScoped'));
+            set(this.item, 'Datacenters', null);
+            break;
+          default:
+            throw e;
+        }
+      }
+    },
```

Comment:
Just curious why you didn't go with the model -> component -> service -> model with this?  I think the current implementation is roughly model -> controller -> service -> dom -> service -> model - and for it to work you have to remember a certain naming system for the inputs (though the addition of throwing would help - I think currently it doesn't? or at least there was a comment that said "we should throw here" :) ). Components would also let you do view-only attributes, and give you hooks to set up the forms instead of doing the `setProperties` override (which I felt was pretty surprising).

Reply:
Woke up with another thought about this 😛 - why not use the action to pass the item and prop name in? Then you wouldn't have to fish around in the DOM and name match from field attrs.

Reply:
Hey, sorry a bit postscript this..

So I had a good experience this week because I was using the standard `name=""` attribute rather than passing the name through as an action argument.

The `fillIn` method in my testing 'just works' if you use `name=""`, not only that but this week whilst I was bringing the old ACL stuff (inc. its testing) back into this, I'd noticed when I came to add/tweak the testing back in that as I was reusing the new 'login' form I needed to change the fillIn value of the old tests to use the new form. I couldn't remember what property I'd used in the new form, but instead of delving into my ember code to find it, I just did an 'Inspect' and there it was in the `name="name of property"` attribute in the element inspector. Probably saved me a few seconds, but a nice detail I got for free by following a standard.